### PR TITLE
qa: avoid hanging indefinitely when umount is stuck

### PR DESCRIPTION
**Pull request by batrick**
Fixes: https://tracker.ceph.com/issues/56694
Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: qa/tasks/cephfs/fuse_mount.py [hunk 1]
```diff
@@ -321,7 +321,7 @@ def umount(self, cleanup=True):
             log.info('Running fusermount -u on {name}...'.format(name=self.client_remote.name))
             stderr = StringIO()
             self.client_remote.run(
-                args=['sudo', 'fusermount', '-u', self.hostfs_mntpt],
+                args=['sudo', 'timeout', '300', 'fusermount', '-u', self.hostfs_mntpt],
                 stderr=stderr, timeout=UMOUNT_TIMEOUT, omit_sudo=False)
```
Comment (idryomov) [inline qa/tasks/cephfs/fuse_mount.py:325]: Both the FUSE umount and the kernel client umount already have `timeout=UMOUNT_TIMEOUT` passed to `run` (also set to 300).  Is that somehow ignored?

Comment (batrick) [reply qa/tasks/cephfs/fuse_mount.py:325]: paramiko ssh command timeouts don't really work the way you would think. I don't know exactly why, but the timeout just shuts down the pipe but doesn't actually cause the command to return (especially if it's in an uninterruptible state).

Comment (idryomov) [reply qa/tasks/cephfs/fuse_mount.py]: It would be good to note this in the commit message which is currently empty.

Comment (batrick) [reply qa/tasks/cephfs/fuse_mount.py]: good point, done

**Comment by dparmar18** [comment]
Looks good, just a quick question, can we try using `wait=False`?

**Pull request closed without merging**